### PR TITLE
Refactored renderer.js -> ui_renderer.js

### PR DESCRIPTION
**Pull request by flawedmatrix**
And a bunch of changes to pieces of the app that integrate with the renderer

### CODE REVIEW

File: lib/static/lib/js/ui_renderer.js [hunk 1]
```diff
@@ -1,19 +1,10 @@
...
 //---------------------------------------------
-(function () {
-var requestAnimationFrame = window.requestAnimationFrame ||
-                            window.mozRequestAnimationFrame ||
-                            window.webkitRequestAnimationFrame ||
-                            window.msRequestAnimationFrame;
 
-window.requestAnimationFrame = requestAnimationFrame;
-})();
-
```
Comment (flawedmatrix) [inline lib/static/lib/js/ui_renderer.js:12]: three.js provides a shim for requestAnimationFrame so we no longer need this.

File: lib/static/lib/js/gs_renderer.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+GSRenderer.prototype.animate = function() {
+    requestAnimationFrame(function() {
+        this.animate();
+    }.bind(this));
+    this.mesh.rotation.x += 0.01;
+    this.mesh.rotation.y += 0.02;
+
+    this.renderer.render(this.scene, this.camera);
+};
```
Comment (flawedmatrix) [inline lib/static/lib/js/gs_renderer.js:26]: This probably needs fixing, but will be handled later in 3D rendering branch.

File: lib/static/lib/js/ui_renderer.js [hunk 3]
```diff
@@ -138,19 +132,31 @@ Renderer.prototype.bindClick = function(clickHandler) {
...
 };
 
-Renderer.prototype.refreshAll = function(game) {
-    this.clear();
-    var nowtime = Date.now();
+UIRenderer.prototype.renderGS = function(gamestate, p_id) {
+    var players = gamestate.players;
+    players.forEach(function (player) {
+        player.units.forEach(function(unit) {
+            this.renderUnit(
```
Comment (flawedmatrix) [inline lib/static/lib/js/ui_renderer.js]: This doesn't need to be multiple lines.

File: lib/static/lib/js/gamestate.js [hunk 4]
```diff
@@ -122,7 +126,8 @@ Unit.prototype.toJSON = function() {
 Unit.prototype.evaluate = function(t) {
     return {
         speed: this.speed,
-        init_pos: this.pos(t),
-        facing: this.facing(t)
+        pos: this.pos(t),
+        facing: this.facing(t),
+        size: this.size
```
Comment (flawedmatrix) [inline lib/static/lib/js/gamestate.js:131]: There needs to be a test for this...

File: jasmine_testing/static/jasmine_testing/specs/GameStateSpec.js [hunk 5]
```diff
@@ -218,12 +218,13 @@ describe("GameState", function() {
             });
             it("should save the position, speed, and facing and nothing else", function() {
                 expect(json).toEqual({
-                    init_pos: {
+                    pos: {
                         x: 0,
                     y: 100
```
Comment (flawedmatrix) [inline jasmine_testing/static/jasmine_testing/specs/GameStateSpec.js]: The formatting is a bit off here.

**Comment by hellochar** [comment]
lg2m

**Pull request merged**